sympy/polys/rings.py

```diff
@@ -3005,3 +3032,222 @@ def sqf_list(f, all=False):
 
     def factor_list(f):
         return f.ring.dmp_factor_list(f)
+
+def monomial_extract(polynomials):
+    """
+    Extracts any common monomial from the polynomials in p.
+
+    Examples
+    ========
+
+    >>> from sympy.polys.rings import monomial_extract
+    >>> from sympy import ZZ, ring
+    >>> R, x, y = ring("x, y", ZZ)
+
+    >>> f = x + y
+    >>> g = x**2*y + x*y
+    >>> polynomials = [f, g]
+    >>> monomial_extract(polynomials)
+    ([x + y, x**2*y + x*y], None)
+
+    >>> f = x**2*y
+    >>> g = x**2*y + x*y
+    >>> polynomials = [f, g]
+    >>> monomial_extract(polynomials)
+    ([x, x + 1], x*y)
+
+    """
+    ring = polynomials[0].ring
+    zero_monoms = ring.zero_monom
+    monoms = chain(*polynomials)
+    monom_gcd = tuple(map(min, zip(*monoms)))
+
+    # If the zero monomial is present, we can immediately return
+    if (any(zero_monoms in poly for poly in polynomials)
+        or monom_gcd == ring.zero_monom):
+        return polynomials, None
+    else:
+        d = ring({monom_gcd: ring.domain.one})
+        p = [pi.exquo(d) for pi in polynomials] # TODO: Use monomial_ldiv
+        return p, d
+
+def gcd_coeffs(polynomials):
+    """
+    Simplify a list of polynomials whose gcd is wanted. Returns a possibly
+    longer list of simpler polynomials having the same gcd as the input.
+
+    This is done by eliminating symbols that can not be part of the gcd
+    because they do not appear in each item of the input. In the output
+    list, all items have exactly the same symbols. The set of those symbols
+    is also returned.
+
+    Examples
+    ========
+
+    >>> from sympy.polys.rings import gcd_coeffs
+    >>> from sympy import ring, ZZ
+    >>> R, x, y = ring("x, y", ZZ)
+
+    >>> f = x**2 - y**2
+    >>> g = x - y
+    >>> polynomials = [f, g]
+    >>> gcd_coeffs(polynomials)
+    ([1], None)
```

Comment:
This example does not in any way help to understand the purpose of this function. We need to have a situation where there are some symbols in common and some that are only in one and not the other and so on.

The comment above about the list being "possibly longer" should also be demonstrated here.